core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -1027,17 +1032,20 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
 
               val result = runLoop(effect.first, stackIndex, stackIndex, currentDepth + 1)
 
-              stackIndex -= 1
-              popStackFrame(stackIndex)
+              if (null eq result)
+                return null
+              else {
```

Comment:
The `else` statement is redundant here

Reply:
I personally prefer using the explicit `else` in this cases, makes it easier to track the code's execution pass, far easier to figure out u're in a conditional branch than spotting a return statement few lines above